palisade-service/src/component-tests/java/uk/gov/gchq/palisade/component/palisade/model/AuditErrorMessageTest.java

```diff
@@ -15,75 +15,64 @@
  */
 package uk.gov.gchq.palisade.component.palisade.model;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.json.JsonTest;
-import org.springframework.boot.test.json.JacksonTester;
-import org.springframework.boot.test.json.JsonContent;
-import org.springframework.boot.test.json.ObjectContent;
 import org.springframework.test.context.ContextConfiguration;
 
 import uk.gov.gchq.palisade.Context;
 import uk.gov.gchq.palisade.service.palisade.model.AuditErrorMessage;
 
-import java.io.IOException;
-import java.util.HashMap;
 import java.util.Map;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.assertAll;
 
 @JsonTest
 @ContextConfiguration(classes = {AuditErrorMessageTest.class})
 class AuditErrorMessageTest {
 
     @Autowired
-    private JacksonTester<AuditErrorMessage> jsonTester;
+    private ObjectMapper mapper;
 
     @Test
     void testContextLoads() {
-        assertThat(jsonTester).isNotNull();
+        assertThat(mapper)
+                .as("Check that the ObjectMapper has been autowired successfully")
+                .isNotNull();
     }
 
     /**
-     * Tests the creation of the message type, AuditErrorMessage using the builder
-     * plus tests the serializing to a Json string and deserializing to an object.
+     * Tests the creation of the AuditErrorMessage using the builder, and tests the serialising to a Json string
+     * and deserialising back to an object.
      *
-     * @throws IOException throws if the {@link AuditErrorMessage} object cannot be converted to a JsonContent.
-     *                     This equates to a failure to serialize or deserialize the string.
+     * @throws JsonProcessingException throws if the {@link AuditErrorMessage} object cannot be converted to a JsonContent.
+     *                                 This equates to a failure to serialise or deserialise the string.
      */
     @Test
-    void testAuditErrorMessageSerializingAndDeserializing() throws IOException {
-        Context context = new Context().purpose("testContext");
-        Map<String, Object> attributes = new HashMap<>();
-        attributes.put("messagesSent", "23");
-
-        AuditErrorMessage auditErrorMessage = AuditErrorMessage.Builder.create()
+    void testAuditErrorMessageSerialisisationAndDeserialisation() throws JsonProcessingException {
```

Comment:
There is the recursive comparison and error message comparison, but it is missing the third one, the equals comparison (uses the isEqualTo() method without the usingRecursiveComparison())  .  This third test will pick-up any objects that do not have a properly implemented equals() and hashCode().

Reply:
updated